.github/workflows/aws-deploy.yml

```diff
@@ -0,0 +1,69 @@
+---
+name: Sample workflow to demonstrate AWS deployment including publishing data to S3 and invalidating a CloudFront distribution cache
+
+on: 
+  workflow_dispatch:
+    inputs:
+        environment:
+            description: "Environment to deploy to"
+            default: "staging"
+            options:
+                - staging
+                - production
+            required: true
+            type: choice
+       
+env:
+  # TODO: Verify what these values are for the allen-cell-animated Github organization
+  AWS_ACCOUNT_ID: ${{ vars.AWS_PUBLIC_DATA_RELEASES_ACCOUNT_ID }}
+  AWS_REGION: ${{ vars.AWS_DEFAULT_REGION }}
+  STAGING_CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.STAGING_CLOUDFRONT_DISTRIBUTION_ID }}
+  STAGING_S3_BUCKET: s3://staging-timelapse.allencell.org
+  PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID }}
+  PRODUCTION_S3_BUCKET: s3://timelapse.allencell.org
+
+permissions:
+    id-token: write  # Required for requesting the JWT and OIDC
+    contents: write  # Required for actions/checkout and OIDC tokens
+
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+
+        # Dynamically set the environment variable based on the input above:
+        environment: ${{ github.event.inputs.environment }}
+
+        steps:
+          - name: Checkout
+            uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
```

Comment:
Any reason why we're using a specific action and not the major version specifier?

Reply:
It's standard practice for us as [suggested by Github](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) to ensure an immutable release. 

I think [this post summarizes the vulnerability](https://www.paloaltonetworks.com/blog/prisma-cloud/unpinnable-actions-github-security/)

Reply:
Thank you! I should probably go back and update our existing actions, then.